src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1796,59 +1790,38 @@ void SolverInterfaceImpl::computeMappings(std::vector<MappingContext> &contexts,
 {
   PRECICE_TRACE();
   using namespace mapping;
-  MappingConfiguration::Timing timing;
   for (impl::MappingContext &context : contexts) {
-    timing      = context.timing;
-    bool mapNow = timing == MappingConfiguration::ON_ADVANCE;
-    mapNow |= timing == MappingConfiguration::INITIAL;
-    bool hasComputed = context.mapping->hasComputedMapping();
-    if (mapNow && not hasComputed) {
+    if (not context.mapping->hasComputedMapping()) {
       PRECICE_INFO("Compute \"{}\" mapping from mesh \"{}\" to mesh \"{}\".",
                    mappingType, _accessor->meshContext(context.fromMeshID).mesh->getName(), _accessor->meshContext(context.toMeshID).mesh->getName());
       context.mapping->computeMapping();
     }
   }
 }
 
-void SolverInterfaceImpl::clearMappings(std::vector<MappingContext> &contexts)
-{
-  PRECICE_TRACE();
-  // Clear non-stationary, non-incremental mappings
-  using namespace mapping;
-  for (impl::MappingContext &context : contexts) {
-    bool isStationary = context.timing == MappingConfiguration::INITIAL;
-    if (not isStationary) {
-      context.mapping->clear();
-    }
-    context.hasMappedData = false;
-  }
-}
-
 void SolverInterfaceImpl::mapWrittenData()
 {
   PRECICE_TRACE();
   computeMappings(_accessor->writeMappingContexts(), "write");
   for (auto &context : _accessor->writeDataContexts()) {
-    if (context.isMappingRequired()) {
+    if (context.hasMapping()) {
       PRECICE_DEBUG("Map write data \"{}\" from mesh \"{}\"", context.getDataName(), context.getMeshName());
       context.mapData();
     }
   }
-  clearMappings(_accessor->writeMappingContexts());
```

Comment:
@fsimonis Wasn't there a reason why we cleared the mapping here? I remember a discussion we once had here when I wanted to remove this line at some point in time...